### PR TITLE
Include dependency values in documentation

### PR DESCRIPTION
**Pull request by marcofranssen**
I have added a flag to the helm-docs script to include the documentation for dependencies.

This will add more complete documentation to https://artifacthub.io/packages/helm/spiffe/spire#values so it is easier for our users to get started and having a complete overview.

### CODE REVIEW

File: charts/spire/charts/spiffe-csi-driver/values.yaml
Comment (faisal-memon) [inline charts/spire/charts/spiffe-csi-driver/values.yaml:11]:
```suggestion
  # -- Overrides the image tag whose default is the chart appVersion.```

File: charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml
Comment (faisal-memon) [inline charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml:19]:
```suggestion
  # -- Overrides the image tag whose default is the chart appVersion
```

File: charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml
Comment (faisal-memon) [inline charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml:128]:
```suggestion
# -- The name of the Kubernetes cluster. (`kubeadm init --service-dns-domain`)
```

File: charts/spire/charts/spire-agent/values.yaml
Comment (faisal-memon) [inline charts/spire/charts/spire-agent/values.yaml:62]:
```suggestion
# -- The name of the Kubernetes cluster. (`kubeadm init --service-dns-domain`)
```

Comment (faisal-memon) [inline charts/spire/charts/spire-agent/values.yaml:62]:
```suggestion
# -- The name of the Kubernetes cluster (`kubeadm init --service-dns-domain`)
```

File: charts/spire/charts/spire-agent/values.yaml
Comment (faisal-memon) [inline charts/spire/charts/spire-agent/values.yaml:64]:
```suggestion
# -- The trust domain to be used for the SPIFFE identifiers
```

**Review by kfox1111 — DISMISSED** [review]
Need to consider the above question before this merges

**Comment by faisal-memon** [comment]
> Need to consider the above question before this merges

I think your suggestions are valid and something we should further discuss. Don't see how it relates to this PR which just fills in descriptions for existing configurables.

File: charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml
Comment (faisal-memon) [inline charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml:88]:
```suggestion
  # -- The log level, valid values are "debug", "info", "warn", and "error"
```

File: charts/spire/charts/spire-agent/values.yaml
Comment (faisal-memon) [inline charts/spire/charts/spire-agent/values.yaml:60]:
```suggestion
# -- The log level, valid values are "debug", "info", "warn", and "error"
```

File: charts/spire/charts/spire-server/values.yaml
Comment (faisal-memon) [inline charts/spire/charts/spire-server/values.yaml:106]:
```suggestion
# -- The log level, valid values are "debug", "info", "warn", and "error"
```

File: charts/spire/values.yaml
Comment (faisal-memon) [inline charts/spire/values.yaml:9]:
```suggestion
    # -- The trust domain to be used for the SPIFFE identifiers
```

File: charts/spire/values.yaml
Comment (faisal-memon) [inline charts/spire/values.yaml:7]:
```suggestion
    # -- The name of the Kubernetes cluster (`kubeadm init --service-dns-domain`)
```

File: charts/spire/charts/spire-server/values.yaml
Comment (faisal-memon) [inline charts/spire/charts/spire-server/values.yaml:28]:
```suggestion
  # -- The name of the service account to use
```

File: charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml
Comment (faisal-memon) [inline charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml:109]:
```suggestion
  # -- The name of the service account to use
```

File: charts/spire/charts/spire-agent/values.yaml
Comment (faisal-memon) [inline charts/spire/charts/spire-agent/values.yaml:25]:
```suggestion
  # -- The name of the service account to use
```

File: charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml
Comment (faisal-memon) [reply charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml:109]: Didnt realize this was a 2 line comment.

**Comment by kfox1111** [comment]
> > Need to consider the above question before this merges
> 
> I think your suggestions are valid and something we should further discuss. Don't see how it relates to this PR which just fills in descriptions for existing configurables.

This PR means all the existing options are set to be synchronized to the README.md of charts/spire

Maybe that is the set of options we want for the base set of values and in that case, your right, this PR is fine. But not sure. That  maybe should be decided on.

I guess this PR is now doing two different things and maybe thats part of the issue too.
https://github.com/spiffe/helm-charts/pull/275/files#diff-68252bdbf9af2b2192f269544f14b135bd58349791567f1c61bf7bd70bfe6a1dR45
does the copy of the values into README.md.

All the rest of the changes are making the values documentation better. No objection to any of that part. Its really great. The only issue is if its the right time to do the nested README values flag. If all the rest of the options were in its own pr, maybe that would decouple things?

File: charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml
Comment (faisal-memon) [inline charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml:128]:
```suggestion
# -- The name of the Kubernetes cluster (`kubeadm init --service-dns-domain`)
```

**Comment by faisal-memon** [comment]
> > > Need to consider the above question before this merges
> > 
> > 
> > I think your suggestions are valid and something we should further discuss. Don't see how it relates to this PR which just fills in descriptions for existing configurables.
> 
> This PR means all the existing options are set to be synchronized to the README.md of charts/spire
> 
> Maybe that is the set of options we want for the base set of values and in that case, your right, this PR is fine. But not sure. That maybe should be decided on.
> 
> I guess this PR is now doing two different things and maybe thats part of the issue too. https://github.com/spiffe/helm-charts/pull/275/files#diff-68252bdbf9af2b2192f269544f14b135bd58349791567f1c61bf7bd70bfe6a1dR45 does the copy of the values into README.md.
> 
> All the rest of the changes are making the values documentation better. No objection to any of that part. Its really great. The only issue is if its the right time to do the nested README values flag. If all the rest of the options were in its own pr, maybe that would decouple things?

I think its good for the user to see all the configurables in the base readme so they don't have to go digging through nested charts to see what can be configured. I agree with your broader point about revisiting documentation and will add this to the agenda for the next working group meeting.

**Comment by kfox1111** [comment]
I think there are plenty of values that we may not want to expose to the average user. For example, spire-agent.socketPath. There is some valid reasons for a user to set it, but its a very advanced use case.

These have all now just been made part of README.md and show up on artifacthub forever in that version, without discussion and over an objection. 

This merging the way it was feels very inappropriate. My own needs have been repeatedly ignored and pushed to the side, and your own have been deemed urgent enough to force through a brand new pr, less then 10 hours old, last minute and cut a release from over an objection simply to have a discussion about it. How can we work as a team when this continues to happen?

**Comment by amoore877** [comment]
> I agree with your broader point about revisiting documentation and will add this to the agenda for the next working group meeting.

What was the fire that we were trying to mitigate that PR merging couldn't also wait for such a full discussion, or continuing that discussion here?
Going back to the theme of slack discussion about the Auto-merge Setting, any project, but especially open source, succeeds based on collaboration and participation; and having a track record of those signals.

Just as a random community member in the early days of SPIRE, if the maintainers merged something I voiced disagreement on without having a full discussion about it, I probably would have pushed my org to abandon adoption and revert to our internal solution.

**Comment by faisal-memon** [comment]
Thanks for the comments @kfox1111 and @amoore877. We want to make sure everyone's feedback is addressed. Merging without doing that was a mistake. Won't be dismissing reviews anymore going forward.

**Pull request merged**
